packages/vue-components/src/components/OmegaForm/OmegaWrapper.vue

```diff
@@ -112,24 +112,26 @@ const localForm = props.form || !props.schema
     {
       ...props,
       onSubmit: typeof props.isLoading !== "undefined"
-        ? ({ value }) =>
-          new Promise<void>((resolve) => {
-            console.log("Submitting form with values:", value, "emitting")
-            instance!.emit("submit", value)
-            // even if the emit would be immediately handled, prop changes are not published/received immediately.
-            const handle = watch(() => props.isLoading, (v) => {
-              if (v) return
-              resolve()
-              handle.stop()
-            })
-          })
+        ? undefined
         : props.onSubmit
     },
     props.omegaConfig
   )
 
 const formToUse = computed(() => props.form ?? localForm!)
```

Comment:
why can't we use defineEmit btw?

Reply:
Because otherwise the defineEmit will cause Vite to override the props onSubmit, and we loose the union that we use to discern in props